lib/matplotlib/backend_managers.py

```diff
@@ -14,6 +18,197 @@
 from matplotlib.rcsetup import validate_stringlist
 import matplotlib.backend_tools as tools
 
+from matplotlib import is_interactive
+from matplotlib import rcParams
+from matplotlib.figure import Figure
+from matplotlib.backends import get_backend, backend as backend_name
+
+
+class FigureManagerEvent(object):
+    """Event for when something happens to this figure manager.
+    i.e. the figure it controls gets closed
+
+    Attributes
+    ----------
+    signal : str
+        The name of the signal.
+
+    figure_manager : FigureManager
+        The figure manager that fired the event.
+    """
+    def __init__(self, signal, figure_manager):
+        self.name = signal
+        self.figure_manager = figure_manager
+
+
+class FigureManager(cbook.EventEmitter):
+    """
+    The FigureManager creates and wraps the necessary components to display a
+    figure, namely the Window, FigureCanvas and Toolbar.  It gets used whenever
+    you want the figure in a standalone window.
+
+    Parameters
+    ----------
+    figure : `matplotlib.figure.Figure`
+        The figure to manage.
+
+    num : int
+        The figure number.
+
+    Attributes
+    ----------
+
+    canvas : `matplotlib.backend_bases.FigureCanvasBase`
+        The GUI element on which we draw.
+
+    figure : `matplotlib.figure.Figure`
+        The figure that holds the canvas
+
+    toolbar : `matplotlib.backend_bases.ToolbarBase`
+        The toolbar used for interacting with the figure.
+
+    window : `matplotlib.backend_bases.WindowBase`
+        The window that holds the canvas and toolbar.
+
+    num : int
+        The figure number.
+    """
+    def __init__(self, figure, num, **kwargs):
+        super(FigureManager, self).__init__(**kwargs)
+        self._backend_name, self._backend = get_backend()
+
+        self.num = num
+        self.figure = figure
+
+        self._is_gui = hasattr(self._backend, 'Window')
+        if not self._is_gui:
+            self.window = None
+            return
```

Comment:
give me copy paste code to show that it doesn't work... ``__init__`` works as a normal method, i.e. it has an implicit ``return``, i.e. ``return None`` at the end...

I think you get confused with ``__new__`` as new does that... not sure what object.__new__ looks like but you get the gist from this:
```python
def __new__(class, *args, **kwargs):
  instance = super().__new__(class, *args, **kwargs)  # calls __init__(self, *args, **kwargs)
  return instance
```
basically:
1. From the code ``MyClass(*args, **kwargs)``
2.`` __new__(MyClass, *args, **kwargs)`` gets called.
3. That magically creates the instance and calls ``__init__`` on it.
4. Returns the instance.